### PR TITLE
ContactRoles: Prevent iteration over string

### PR DESCRIPTION
**Pull request by dontub**
APIv3 returns `'0'` on `getvalue` action if no value is set. In that case it was tried to iterate over a string.

### CODE REVIEW

File: CRM/Remotetools/ContactRoles.php [hunk 1]
```diff
@@ -69,14 +69,11 @@ public static function getRoles($contact_id)
...
-            CRM_Remotetools_CustomData::resolveCustomFields($query);
-            $roles = civicrm_api3('Contact', 'getvalue', [
-                'id'     => $contact_id,
-                'return' => $roles_field
-            ]);
+            $roles = Contact::get(FALSE)
+              ->addSelect('remote_contact_data.remote_contact_roles')
+              ->addWhere('id', '=', $contact_id)
+              ->execute()
+              ->first()['remote_contact_data.remote_contact_roles'] ?? [];
```
[the remark this answers is not in the document]
Comment (dontub) [reply CRM/Remotetools/ContactRoles.php:76]: If `$contact_id` must be a valid contact ID then `single()` would be fine.

[the remark this answers is not in the document]
Comment (dontub) [reply CRM/Remotetools/ContactRoles.php:76]: Names and labels are now fetched via the APIv4 action, so the call of `getAllRoles()` isn't necessary, anymore.

[the remark this answers is not in the document]
Comment (jensschuppe) [reply CRM/Remotetools/ContactRoles.php:76]: Well, it should be a valid contact ID, and `first()` won't return anything different than `single()` in this case …

[the remark this answers is not in the document]
Comment (dontub) [reply CRM/Remotetools/ContactRoles.php]: Changed `first()` to `single()`.

**Review by jensschuppe — APPROVED** [review]

**Pull request merged**